code/datums/status_effects/buffs.dm

```diff
@@ -450,3 +450,96 @@
 /datum/status_effect/exercised/Destroy()
 	. = ..()
 	STOP_PROCESSING(SSprocessing, src)
+
+//Hippocratic Oath: Applied when the Rod of Asclepius is activated.
+/datum/status_effect/hippocraticOath
+	id = "Hippocratic Oath"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = -1
+	tick_interval = 25
+	examine_text = "<span class='notice'>They seem to have an aura of healing and helpfulness about them.</span>"
+	alert_type = null
+	var/hand = null
+	var/deathTick = 0
+
+/datum/status_effect/hippocraticOath/on_apply()
+	//Makes the user passive, it's in their oath not to harm!
+	owner.add_trait(TRAIT_PACIFISM, "hippocraticOath")
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+	H.add_hud_to(owner)
+	return ..()
+
+/datum/status_effect/hippocraticOath/on_remove()
+	owner.remove_trait(TRAIT_PACIFISM, "hippocraticOath")
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+	H.remove_hud_from(owner)
+
+/datum/status_effect/hippocraticOath/tick()
+	if(owner.stat == DEAD)
+		if(deathTick < 4)
+			deathTick += 1
+		else
+			owner.visible_message("[owner]'s soul is absorbed into the rod, releaving the previous snake of it's duty.")
+			var/mob/living/simple_animal/hostile/retaliate/poison/snake/healSnake = new /mob/living/simple_animal/hostile/retaliate/poison/snake(owner.loc)
+			var/list/chems = list("bicaridine", "salbutamol", "kelotane", "antitoxin")
+			healSnake.poison_type = pick(chems)
+			healSnake.name = "Asclepius's Snake"
+			healSnake.real_name = "Asclepius's Snake"
+			healSnake.desc = "A mystical snake previously trapped upon the Rod of Asclepius, now freed of it's burden. Unlike the average snake, it's bites contain chemicals with minor healing properties."
+			new /obj/effect/decal/cleanable/ash(owner.loc)
+			new /obj/item/rod_of_asclepius(owner.loc)
+			qdel(owner)
+	else
+		if(iscarbon(owner))
+			var/mob/living/carbon/itemUser = owner
+			var/obj/item/heldItem = itemUser.get_item_for_held_index(hand)
+			if(heldItem == null || heldItem.type != /obj/item/rod_of_asclepius) //Checks to make sure the rod is still in their hand
+				var/obj/item/rod_of_asclepius/newRod = new /obj/item/rod_of_asclepius(itemUser.loc)
+				newRod.flags_1 = NODROP_1 | DROPDEL_1
+				newRod.desc = "A short wooden rod with a mystical snake inseparably gripping itself and the rod to your forearm. It flows with a healing energy that disperses amongst yourself and those around you. "
+				newRod.icon_state = "asclepius_active"
+				newRod.activated = TRUE
```

Comment:
give the rod an activate proc so if you have to change activation you don't have to look across the file for each time it gets activated.